modules/rest/src/main/java/org/apache/ignite/internal/rest/RestModule.java

```diff
@@ -17,136 +17,73 @@
-import java.util.Map;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.CompletionException;
+import java.util.function.Consumer;
 import org.apache.ignite.configuration.schemas.rest.RestConfiguration;
 import org.apache.ignite.configuration.schemas.rest.RestView;
-import org.apache.ignite.configuration.validation.ConfigurationValidationException;
 import org.apache.ignite.internal.configuration.ConfigurationManager;
 import org.apache.ignite.internal.configuration.ConfigurationRegistry;
 import org.apache.ignite.internal.manager.IgniteComponent;
-import org.apache.ignite.internal.rest.netty.RestApiHttpRequest;
-import org.apache.ignite.internal.rest.netty.RestApiHttpResponse;
 import org.apache.ignite.internal.rest.netty.RestApiInitializer;
-import org.apache.ignite.internal.rest.presentation.ConfigurationPresentation;
-import org.apache.ignite.internal.rest.presentation.hocon.HoconPresentation;
 import org.apache.ignite.internal.rest.routes.Router;
+import org.apache.ignite.internal.rest.routes.SimpleRouter;
 import org.apache.ignite.lang.IgniteException;
 import org.apache.ignite.lang.IgniteLogger;
 import org.apache.ignite.network.NettyBootstrapFactory;
+import org.apache.ignite.rest.RestHandlersRegister;
+import org.apache.ignite.rest.Routes;
 
 /**
  * Rest module is responsible for starting a REST endpoints for accessing and managing configuration.
  *
- * <p>It is started on port 10300 by default but it is possible to change this in configuration itself. Refer to default config file in
+ * <p>It is started on port 10300 by default, but it is possible to change this in configuration itself. Refer to default config file in
  * resources for the example.
  */
-public class RestModule implements IgniteComponent {
-    /** Default port. */
-    public static final int DFLT_PORT = 10300;
-
-    /** Node configuration route. */
-    private static final String NODE_CFG_URL = "/management/v1/configuration/node/";
-
-    /** Cluster configuration route. */
-    private static final String CLUSTER_CFG_URL = "/management/v1/configuration/cluster/";
-
-    /** Path parameter. */
-    private static final String PATH_PARAM = "selector";
-
+public class RestModule implements RestHandlersRegister, IgniteComponent {
     /** Ignite logger. */
     private final IgniteLogger log = IgniteLogger.forClass(RestModule.class);
 
     /** Node configuration register. */
     private final ConfigurationRegistry nodeCfgRegistry;
 
-    /** Presentation of node configuration. */
-    private final ConfigurationPresentation<String> nodeCfgPresentation;
-
-    /** Presentation of cluster configuration. */
-    private final ConfigurationPresentation<String> clusterCfgPresentation;
-
     /** Netty bootstrap factory. */
     private final NettyBootstrapFactory bootstrapFactory;
 
+    private final SimpleRouter router = new SimpleRouter();
+
     /** Netty channel. */
     private volatile Channel channel;
```

Comment:
I would suggest to re-write this logic using the busy lock approach

Reply:
This should probably be implemented in a separate issue.
I filed a ticket about this: https://issues.apache.org/jira/browse/IGNITE-16636